tests/e2e/matrix/conduit.go

```diff
@@ -0,0 +1,202 @@
+
+	cmd := exec.Command(matrixBinPath)
+	cmd.Env = append(os.Environ(),
+		"CONDUIT_DATABASE_PATH="+dbDir,
+		"CONDUIT_PORT="+strconv.Itoa(port),
+	)
+	cmd.Stdout = w
+	cmd.Stderr = w
+
+	if err := cmd.Start(); err != nil {
+		return nil, nil, fmt.Errorf("failed to start matrix server (%d): %w", cmd.Process.Pid, err)
+	}
+
+	m := &MatrixServer{
+		logger:               logger,
+		pid:                  cmd.Process.Pid,
+		matrixDir:            matrixDir,
+		client:               client,
+		networkFeeBotAddress: crypto.PubkeyToAddress(networkFeeKey.PublicKey),
+	}
+
+	if err := m.awaitReady(ctx); err != nil {
+		return nil, nil, fmt.Errorf("failed to wait for matrix (%d) to be ready: %w", cmd.Process.Pid, err)
+	}
+
+	// TODO @evlekht move this to e2e ? wrap StartNewMatrixServer func with that logic? resolve when ASB is introduced in e2e
+	m.networkFeeCMAccountAddress, _, err = networkClient.CreateCMAccount(ctx, networkFeeKey)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create cm account: %w", err)
+	}
+
+	if err := networkClient.Transfer(ctx, networkClient.PrefundedKeys()[0], m.networkFeeBotAddress, e2eCommon.DefaultCMAccountOwnerFunds); err != nil {
+		return nil, nil, fmt.Errorf("failed to transfer funds to cm account owner: %w", err)
+	}
+
+	if err := networkClient.AddBotToCMAccount(ctx, m.networkFeeCMAccountAddress, networkFeeKey, m.networkFeeBotAddress); err != nil {
+		return nil, nil, fmt.Errorf("failed to add bot to CM account: %w", err)
+	}
+
+	logger.Infof("Matrix server (pid %d) started", cmd.Process.Pid)
+
+	errChan := make(chan error)
+	go func() {
+		err := <-process.ListenForProcessError(cmd)
+		if err != nil {
+			errChan <- fmt.Errorf("matrix server (pid %d) failed: %w", m.pid, err)
+		}
+		close(errChan)
+	}()
+
+	return m, errChan, nil
+}
+
+// Conduit.
+// Not safe for concurrent use.
+type MatrixServer struct {
+	logger                     *zap.SugaredLogger
+	pid                        int
+	matrixDir                  string
+	client                     *mautrix.Client
+	networkFeeBotAddress       common.Address
+	networkFeeCMAccountAddress common.Address
+}
+
+func (m *MatrixServer) Host() *url.URL {
+	return m.client.HomeserverURL
+}
+
+func (m *MatrixServer) Stop(ctx context.Context) error {
+	if m == nil {
+		return nil
+	}
+	if err := process.StopProcess(ctx, m.pid); err != nil {
+		return fmt.Errorf("failed to stop matrix server process with pid %d: %w", m.pid, err)
+	}
+	m.logger.Infof("Matrix server (pid %d) stopped", m.pid)
+	return nil
+}
+
+func (m *MatrixServer) NetworkFeeRecipientBotAddress() common.Address {
+	return m.networkFeeBotAddress
+}
+
+func (m *MatrixServer) NetworkFeeRecipientCMAccountAddress() common.Address {
+	return m.networkFeeCMAccountAddress
+}
+
+// Will try to do periodic login
+func (m *MatrixServer) awaitReady(ctx context.Context) error {
+	cryptoHelper, err := cryptohelper.NewCryptoHelper(m.client, []byte("meow"), path.Join(m.matrixDir, "client-db"))
```

Comment:
"meow" 😹

Reply:
do you want it changed?

Reply:
No worries - there is no need to change - just found it funny 😸

Reply:
😸